fips202/fips202x4.h

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+#ifndef FIPS_202X_H
+#define FIPS_202X_H
+
+#ifndef KECCAK_WAY
+#define KECCAK_WAY 4
+#endif
+
+#include <stdint.h>
+
+typedef struct
+{
+    uint64_t ctx[25 * KECCAK_WAY];
+} keccakx4_state;
+
+uint64_t *keccakx_get_lane_state(keccakx4_state *state, size_t index);
+
+int shake128x4_absorb(keccakx4_state *state,
+                      const uint8_t *in0,
+                      const uint8_t *in1,
+                      const uint8_t *in2,
+                      const uint8_t *in3,
+                      size_t inlen);
+
+int shake256x4_absorb(keccakx4_state *state,
+                      const uint8_t *in0,
+                      const uint8_t *in1,
+                      const uint8_t *in2,
+                      const uint8_t *in3,
+                      size_t inlen);
+
+
+int shake128x4_squeezeblocks(uint8_t *out0,
+                             uint8_t *out1,
+                             uint8_t *out2,
+                             uint8_t *out3,
+                             size_t nblocks,
+                             keccakx4_state *state);
+
+int shake256x4_squeezeblocks(uint8_t *out0,
+                             uint8_t *out1,
+                             uint8_t *out2,
+                             uint8_t *out3,
+                             size_t nblocks,
+                             keccakx4_state *state);
+
+/*
+ * Squeezes a single lane in Keccak 4-way
+ */
+int shake256x1_squeezeblocks(uint8_t *out,
```

Comment:
I propose to rename this to `shake256x4_squeezeblocks_single`. 
Also I don't think referring to this as a `lane` is correct, because a lane in Keccak is well defined. 
I'd just call it `single Keccak state`

Reply:
That's good idea. update in https://github.com/pq-code-package/mlkem-c-aarch64/pull/62/commits/6c716d18f1aada08abcd445c905fb77937dce1d5